src/plugins/dashboard/public/application/utils/get_nav_actions.tsx

```diff
@@ -312,23 +316,25 @@ export const getNavActions = (
         ? createDashboardEditUrl(savedDashboard.id)
         : DashboardConstants.CREATE_NEW_DASHBOARD_URL;
 
-      dashboardContainer?.updateAppStateUrl?.(pathname, false);
+      currentContainer?.updateAppStateUrl?.({ replace: false, pathname });
 
+      const newStateContainer: { [key: string]: any } = {};
       // This is only necessary for new dashboards, which will default to Edit mode.
-      stateContainer.transitions.set('viewMode', ViewMode.VIEW);
+      newStateContainer.viewMode = ViewMode.VIEW;
```

Comment:
The comment above mentions that this should be edit mode but the code reads View mode. since the tests pass i assume the comment is incorrect?

Reply:
Its a little bit confusing but this logic's goal is to always force it to view mode. And that it is to handle the case where a new dashboard is created it will be created in EDIT mode. So this logic will make sure to turn it to VIEW mode.